verticapy/tests_new/plotting/base_test_files.py

```diff
@@ -0,0 +1,2253 @@
+        """
+        Store labels for X,Y,Z axis to check.
+        """
+        return [self.COL_NAME_VDF_1, "density"]
+
+    def create_plot(self):
+        """
+        Create the plot
+        """
+        return (
+            self.data.bar,
+            {"columns": self.COL_NAME_VDF_1},
+        )
+
+    def col_name_param():
+        """
+        Get column value to pass as pytest parameter
+        """
+        return "0"
+
+    @pytest.mark.parametrize(
+        "of_col, method", [(col_name_param(), "min"), (col_name_param(), "max")]
+    )
+    def test_properties_output_type_for_all_options(
+        self,
+        dummy_dist_vd,
+        plotting_library_object,
+        of_col,
+        method,
+    ):
+        """
+        Test of and method combination
+        """
+        # Arrange
+        # Act
+        result = dummy_dist_vd[self.COL_NAME_VDF_1].bar(
+            of=of_col,
+            method=method,
+        )
+        # Assert - checking if correct object created
+        assert isinstance(result, plotting_library_object), "Wrong object created"
+
+
+class VDCBarhPlot(BasicPlotTests):
+    """
+    Testing different attributes of HHorizontal Bar plot on a vDataColumn
+    """
+    # Testing variables
+    COL_NAME = "check 2"
+    COL_NAME_2 = "check 1"
+
+    @pytest.fixture(autouse=True)
+    def data(self, dummy_vd):
+        """
+        Load test data
+        """
+        self.data = dummy_vd
+
+    @property
+    def cols(self):
+        """
+        Store labels for X,Y,Z axis to check.
+        """
+        return [self.COL_NAME, "density"]
+
+    def create_plot(self):
+        """
+        Create the plot
+        """
+        return (
+            self.data[self.COL_NAME].barh,
+            {},
+        )
+
+    @pytest.mark.parametrize(
+        "max_cardinality, method", [(1, "mean"), (1, "max"), (2, "sum")]
+    )
+    def test_properties_output_type_for_all_options(
+        self,
+        dummy_vd,
+        method,
+        plotting_library_object,
+        max_cardinality,
+    ):
+        """
+        Test max_cardinality and method combination
+        """
+        # Arrange
+        # Act
+        result = dummy_vd[self.COL_NAME].barh(
+            method=method,
+            of=self.COL_NAME_2,
+            max_cardinality=max_cardinality,
+        )
+        # Assert - checking if correct object created
+        assert isinstance(result, plotting_library_object), "Wrong object created"
+
+
+class VDFBarhPlot(VDFBarPlot):
+    """
+    Testing different attributes of HHorizontal Bar plot on a vDataFrame
+    """
+
+    def create_plot(self):
+        """
+        Create the plot
+        """
+        return (
+            self.data.barh,
+            {"columns": self.COL_NAME_VDF_1},
+        )
+
+
+# Testing variables
+
+
+class VDCBoxPlot(BasicPlotTests):
+    """
+    Testing different attributes of Box plot on a vDataColumn
+    """
+
+    COL_NAME_1 = "0"
+    COL_NAME_2 = "binary"
```

Comment:
I don't see usage of `COL_NAME_2`

Reply:
Never mind, it is used in the subclass.

Reply:
it is used at line 464 ... as the class is inherited